python/cuml/cuml/datasets/utils.py

```diff
@@ -13,23 +13,25 @@
 # limitations under the License.
 #
 
+from cuml.internals.global_settings import GlobalSettings
 from cuml.internals.safe_imports import gpu_only_import
 
 cp = gpu_only_import("cupy")
 
 
 def _create_rs_generator(random_state):
     """
-    This is a utility function that returns an instance of CuPy RandomState
+    This is a utility function that returns an instance of CuPy/numpy
+    RandomState depending on the current globally-selected device type
     Parameters
     ----------
-    random_state : None, int, or CuPy RandomState
-        The random_state from which the CuPy random state is generated
+    random_state : None, int, or RandomState
+        The random_state from which the random state is generated
     """
 
     if isinstance(random_state, (type(None), int)):
-        return cp.random.RandomState(seed=random_state)
-    elif isinstance(random_state, cp.random.RandomState):
+        return GlobalSettings().xpy.random.RandomState(seed=random_state)
+    elif isinstance(random_state, GlobalSettings().xpy.random.RandomState):
         return random_state
     else:
```

Comment:
I don't understand the global settings thing well enough (and this is an internal function), but I think it can happen that `xpy` is cupy and the caller passes in a numpy random state. Maybe useful to have an explicit case for this to tell the caller that "yes you passed a random state, but the wrong kind"